execution/sym_region_frag_machine.ml

```diff
@@ -1578,12 +1622,29 @@ struct
 		      self#target_solve_single offset cond_v wd
 		  | None -> ())
 	   | _ -> ());
-	(match ty with
-	   | V.REG_8 -> self#store_byte_region r addr value
-	   | V.REG_16 -> self#store_short_region r addr value
-	   | V.REG_32 -> self#store_word_region r addr value
-	   | V.REG_64 -> self#store_long_region r addr value
-	   | _ -> failwith "Unsupported type in memory move")
+	let table_store_status =
+	  if is_table_store = true then (
+	    if addr <> 0L then 
+	      failwith "table store to symbolic region with non-zero concrete offset, panic!";
+	    match self#decide_maxval "Store" off_exp 0L with
+	    | None -> false
+	    | Some maxval -> 
+	      self#table_store cloc r off_exp 
+		(D.to_symbolic_32 (self#eval_int_exp_simplify addr_e)) 
+		maxval ty value)
+	  else (
+	    if cloc <> 0L then 
+	      failwith "table store to symbolic region with non-zero cloc, panic!";
+	    false )
+	in
+	if table_store_status <> true then
```

Comment:
When branching on a boolean value "b", I usually prefer to say "if b" or "if not b", rather than using "= true", "= false", "<> true", or "<> false". It's just an issue of style, but in this case I find "<> true" to be the least readable, because the negation is least obvious when it's embedded in "<>" than when there is "not" or "false".